src/scripts.js

```diff
@@ -287,16 +297,23 @@ function deselectTag() {
 // ----- Searching -----
 
 function searchRecipesByName(search) {
-  const nameFilteredRecipes = store.recipeRepo.searchByName(search);
-  if (nameFilteredRecipes.length > 0) {
+  if (allRecipesTab.checked && search.length > 0) {
+    let nameFilteredRecipes = store.recipeRepo.searchByName(search)
+    updateRecipeDisplay(nameFilteredRecipes);
+    changeSearchButton(clearAllRecipeSearchButton, searchAllRecipesButton);
+  } 
+  else if (cookbookTab.checked && search.length > 0) {
+    let nameFilteredRecipes = store.user.favoriteRecipeRepo.searchByName(search)
     updateRecipeDisplay(nameFilteredRecipes);
-    changeSearchButton();
+    changeSearchButton(clearCookbookSearchButton, searchCookbookButton);
   }
 }
 
-function changeSearchButton() {
+function changeSearchButton(clearButton, searchButton) {
   cookbookSearchBar.value = '';
   allRecipesSearchBar.value = '';
+  show(clearButton)
+  hide(searchButton)
```

Comment:
@Adelle-Pitsas I haven't been a part of the search so not sure how hard would this be! But could we make that it so it doesn't clear out the`keyword` until you click the `clear` button? That way the user knows they are clearing out the search for that keyword. Right now it defaults to the original gray search message as soon as you search

Reply:
Yeah, can definitely do that!

Reply:
Updated so there is an event listener on the clear button to clear the search bar when clicked. Added a new helper function `clearSearchBar` on `line 326`

Reply:
Thank you! Was this change in the PR? I pulled and checkout into the branch but don't see the clearSearchBar on line 326 and it clears the keyword on search. I may pulled down wrong on my end, but wanted to check!

Reply:
Awesome! Thank you! Just one last thing - I think we need to add the `clearSearchBar` function for whenever the user clicks on the `All recipes tab` or the `cookbook tab`.  Right now if they don't follow our user flow of hitting the clear button, but instead click the tab then the keyword will be there when they go back. I'm pretty sure we just need to add it so when they click the tab it runs the clear search bar. 

https://user-images.githubusercontent.com/108706408/197537843-d7593f9d-af30-410d-bba2-4f0891d64274.mp4

Reply:
Good catch! Just added it to the event listener function.